docs/spec_draft.md

```diff
@@ -1267,6 +1268,137 @@ More formally, `result[i0, ..., iR-1]` is equal to:
+     is to skip the holes in dilated window.
+
+The evaluation order of the reduction function is arbitrary and may be
+non-deterministic. Therefore, the reduction function should not be overly
+sensitive to reassociation. See the discussion about associativity in the
+context of [stablehlo.reduce](#stablehloreduce) for more details.
+
+### Inputs
+
+| Name                | Type                                             | Constraints |
+|---------------------|--------------------------------------------------|-------------|
+| `inputs`            | variadic number of tensors of any supported type | (C1), (C3)  |
+| `init_values`       | variadic number of scalars of any supported type | (C1), (C2)  |
+| `window_dimensions` | 1-dimensional tensor constant type `si64`        | (C4)        |
+| `window_strides`    | 1-dimensional tensor constant type `si64`        | (C5)        |
+| `base_dilations`    | 1-dimensional tensor constant type `si64`        | (C6)        |
+| `window_dilations`  | 1-dimensional tensor constant type `si64`        | (C7)        |
+| `padding`           | 2-dimensional tensor constant type `si64`        | (C8)        |
+| `body`              | `function`                                       | (C9)        |
+
+### Outputs
+
+| Name     | Type                                              | Constraints |
+|----------|---------------------------------------------------|-------------|
+| `results` | variadic number of tensors of any supported type | (C1), (C8), (C9) |
+
+### Constraints
+
+  * (C1) size(`inputs`) $=$ size(`init_values`) $=$ size(`results`) $=$ N and
+         N $\ge$ 1
+
+  * (C2) All `init_values` values have to form an identity under reduction
+         `body`. The behavior of the operation is undefined otherwise.
+
+  * (C3) All `inputs` have the same shape.
+
+  * (C4) On `window_dimensions`
+    * size(`window_dimensions`) $=$ rank(`inputs[k]`) for any k $\in$ [0, N).
+    * `window_dimensions[i]` $\gt 0$ for all i $\in$ [0, size(window_dimensions))
+
+  * (C5) On `window_strides` (optional)
+    * Default value of 1 for each dimension of inputs[k] for any k $\in$ [0, N).
+    * size(`window_strides`) $=$ rank(`inputs[k]`) for any k $\in$ [0, N).
+    * `window_strides[i]` $\gt 0$ for all i $\in$ [0, size(window_strides))
+
+  * (C6) On `base_dilations` (optional)
+    * Default value of 1 for each dimension of inputs[k] for any k $\in$ [0, N).
+    * size(`base_dilations`) $=$ rank(`inputs[k]`) for any k $\in$ [0, N).
+    * `base_dilations[i]` $\gt 0$ for all i $\in$ [0, size(base_dilations))
+
+  * (C7) On `window_dilations` (optional)
+    * Default value of 1 for each dimension of inputs[k] for any k $\in$ [0, N).
+    * size(`window_dilations`) $=$ rank(`inputs[k]`) for any k $\in$ [0, N).
+    * `window_dilations[i]` $\gt 0$ for all i $\in$ [0, size(window_dilations))
+
+  * (C8) On `padding`(optional)
+    * Default value of (0,0) for each dimension of inputs[k] for any k $\in$ [0, N).
```

Comment:
Why is the default value specifically a two value tuple (0, 0)? Perhaps just 0-equivalent value? (e.g. false for boolean and (0.0, 0.0) for complex)

Reply:
The padding tuple (a, b) for a particular dimension d represents the a number of padding values to be added  at the low-end (next to index 0) and b number of padding values to be added at the high-end (next to the
highest index) of dimension `d` respectively. For this op, the semantics is that the padding should not affect the result of computation: So either we can skip them or use int_value as the padding value.

Reply:
The default value (0,0) means no padding added at either ends.

Reply:
Ah I think there was a confusion regarding this. First, I didn't realize that padding could be 2-dimensional, so now I understand what (0, 0) was meant for. However, for this specific case, I was referring to what the default of the padded tensor would be. For example, if we pad a 2x2 matrix with (1, 1), what is the filled value to the cells that was added as a result of the padding?
```
// operand
[-1, -1]
[-1, -1]

// padded operand
[?, ?]
[-1, -1]
[-1, -1]
[?, ?]
```
What is the value of `?` is this the same as `init_values[k]`? This could be related to the actual implementation, but since if we default this to 0-equivalent, doing a reduction operation like max on a 4x1 window of [0, -1, -1, 0] where 0 is the "default" padded value, then would returning a 0 be a side-effect or intentional?

Reply:
>> This could be related to the actual implementation
Yes. If the implementation decides on **not** to skip those values and rather include them in computation, then `init_value` should be the choice. IOW, having `init_value` for padding and dilation holes will have the same effect as skipping those.

Reply:
Makes sense, as you mentioned, for product reduction, since the `init_value` would most likely be 1, the dilation holes having the value 1 would have the same effect as skipping it.